### ISSUE TITLE
Imprecise description of `Changed` event behavior in the class `MyList<T>`

### ISSUE THREAD

**Issue by rogeriomc** [dotnet/docs#33327]
**Issue description**

The first phrase of the the third paragraph of the **Events** topic is imprecise: 

>The `MyList<T>` class declare a single event member called `Changed`, which indicates that a new item has been added to the list.

This description of the `Changed` event behavior is incomplete since the event is also triggered when some item is changed via the indexer set accessor of the generic class `MyList<T>`. I have one suggestion to complement the cited phrase and another to modify the code to be more confident with the event name `Changed`.

1. Complement the first phrase of the the third paragraph of the **Events** topic to something similar to the complement that follow in bold:
> The `MyList<T>` class declare a single event member called `Changed`, which indicates that a new item has been added to the list **or a list item has been changed using the indexer set accessor**.

2. Modify the indexer set accessor code of the `MyList<T>` class to only call the `OnChange()` virtual method if the value of the item has really been changed as follow:
```
public T this[int index]
{
   get => _items[index];
   set
   {
      if (!object.Equals(_items[index], value))
      {
         _items[index] = value;
         OnChanged();
      }
   }
}
```

**Target framework**

- [X] .NET Core
- [X] .NET Framework
- [X] .NET Standard

<details>
<summary><strong>.NET SDK 7.0.101 on Windows 10.0.22621</strong> or <strong>Microsoft Visual Studio 2022 Community 2022 (64-bit) - Current Version 17.4.3</strong></summary>

```console
<replace>
```
</details>

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 3148b5e9-550e-67c9-eb13-907cb78f7fe7
* Version Independent ID: c723989f-2ff6-389d-dcb4-f55d97d14501
* Content: [A tour of C# - The building blocks of C# programs](https://learn.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/program-building-blocks#events)
* Content Source: [docs/csharp/tour-of-csharp/program-building-blocks.md](https://github.com/dotnet/docs/blob/main/docs/csharp/tour-of-csharp/program-building-blocks.md)
* Product: **dotnet-csharp**
* Technology: **csharp-get-started**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
Hi @rogeriomc 

These are great suggestions. I've added this issue to our backlog, and added the "help wanted" label if anyone wants to submit a PR with these changes before we next update this article.

**Comment by BillWagner** [comment]
Fixed in #33359
